packages/syft/src/syft/core/tensor/ancestors.py

```diff
@@ -479,12 +495,22 @@ def _private(
         if max_vals.shape != self.child.shape:
             max_vals = lazyrepeatarray(max_vals, self.child.shape)
 
-        self.replace_abstraction_top(
-            tensor_type=_PhiTensor(),
-            child=self.child,
-            min_vals=min_vals,
-            max_vals=max_vals,
-            data_subjects=data_subjects,  # type: ignore
-        )  # type: ignore
+        unique_data_subjects = len(data_subjects.sum())
```

Comment:
Q: I think the DSA works at data point level, Do we consider uniqueness based on the data point or the row of the tensor?
Ex: 
Data  = np.array([[1,2,],
                              [3,4]])

data_subjects = np.array( [ [DSA("natsu"), DSA("natsu")] ] ,
                                              [DSA("natsu"),DSA("zerek")]]
Since the data subjects for a data point are all unique, this is a PhiTensor?
but the above code would consider it as a gamma tensor.

Maybe we could check for uniqueness at the data point level?
Gamma Tensor Data Subjects:

DataSubjects: np.array([[DSA("natsu","zerek"), DSA("grey")]
                                            [DSA("mavis") ,DSA("erza")]]